SWAN/make_1D_profiles_WZ_productie_buffermethod.py

```diff
@@ -0,0 +1,322 @@
+        df_xy['z'] = np.zeros_like(df_xy['x'])
+
+        df_xy_ahn = pd.DataFrame({'x':list(zip(*chain_coords))[0],'y':list(zip(*chain_coords))[1]})
+        df_xy_ahn['z'] = np.zeros_like(df_xy_ahn['x'])
+
+    
+        #%% drape on xy for bathy and ahn
+        interpolate.interpolate_xyz_on_xy(df_xyz,df_xy)
+        interpolate.interpolate_xyz_on_xy(df_xyz_ahn, df_xy_ahn)
+
+        # if ahn-data is available, replace values from bathy with the values from ahn
+        df_xy.loc[df_xy_ahn['z'].notnull().values] = df_xy_ahn.loc[df_xy_ahn['z'].notnull().values]
+
+        # make buffer for interpolation on edge bathy and ahn if enough ahn datapoints
+
+        buff_data_outside = df_xy_ahn['z'][::-1].notnull().idxmax()
+        if buff_data_outside > 6:
+            df_xy.loc[(buff_data_outside-5):(buff_data_outside+5), 'z'] = np.nan
+        else:
+            print('Not enough datapoints to create buffer for interpolation on edge of data')
+
+        
+
+
+        # func for last processes of df_xy
+        # def finish_df(df_xy, max_len):
+        #     last_valid_ind = df_xy['z'].last_valid_index()
+        #     df_xy = df_xy.iloc[:last_valid_ind]
+           
+        #     # calculate distance and fill out nans
+        #     df_xy['distance'] = np.sqrt((df_xy['x']-df_xy['x'][0])**2+(df_xy['y']-df_xy['y'][0])**2)
+        #     df_xy['zfilled'] = df_xy['z'].interpolate()
+            
+        #     # limit to specify max length
+        #     df_xy = df_xy.drop(df_xy[df_xy['distance'] > max_len].index)
+            
+        #     # remove any remaining nans
+        #     nanlist = list(np.where(df_xy['zfilled'].isnull())[0])
+        #     for ix in nanlist:
+        #         if ix == 0:
+        #             df_xy['zfilled'][ix] = df_xy['zfilled'][nanlist[-1]+1]
+        #         else:
+        #             df_xy['zfilled'][ix] = df_xy['zfilled'][nanlist[-1]+1]
+        #     return df_xy
+
+        # df_xy = finish_df(df_xy, max_len)
+        # df_xy_ahn = finish_df(df_xy_ahn, max_len)
+        # df_xy_bathy = finish_df(df_xy_bathy,max_len)
+
+        
+
+        # #%% finish df_xy dataframe
+        # limit to last valid index of z
+        last_valid_ind = df_xy['z'].last_valid_index()
+        df_xy = df_xy.iloc[:last_valid_ind]
+           
+        # calculate distance and fill out nans
+        df_xy['distance'] = np.sqrt((df_xy['x']-df_xy['x'][0])**2+(df_xy['y']-df_xy['y'][0])**2)
+        df_xy['zfilled'] = df_xy['z'].interpolate()
+        # x_overlap = df_xy['distance'][buff_data_outside]
+        
+        # limit to specify max length
+        df_xy = df_xy.drop(df_xy[df_xy['distance'] > max_len].index)
+        
+        # remove any remaining nans
+        nanlist = list(np.where(df_xy['zfilled'].isnull())[0])
+        for ix in nanlist:
+            if ix == 0:
+                df_xy['zfilled'][ix] = df_xy['zfilled'][nanlist[-1]+1]
```

Comment:
i don't get this line: 
1. it's there twice
2. I expected the `nanlist` to be `ix`. now each nan is filled with the value after the last. I would expect each nan value is filled with the value next to it (`ix+1`)